torch/testing/_internal/distributed/rpc/cuda_distributed_rpc_test.py

```diff
@@ -0,0 +1,95 @@
+import os
+import sys
+import time
+import torch
+import torch.distributed.autograd as autograd
+import torch.distributed.rpc as rpc
+import torch.nn as nn
+
+from torch.testing._internal.distributed.rpc.rpc_agent_test_fixture import (
+    RpcAgentTestFixture,
+)
+from torch.testing._internal.dist_utils import (
+    dist_init,
+)
+
+from torch.testing._internal.common_distributed import (
+    TEST_SKIPS,
+)
+
+
+class MyModule(nn.Module):
+    def __init__(self, device, comm_mode):
+        super().__init__()
+        self.device = device
+        self.linear = nn.Linear(1000, 1000).to(device)
+        self.comm_mode = comm_mode
+
+    def forward(self, x):
+        # x.to() is a no-op if x is already on self.device
+        y = self.linear(x.to(self.device))
+        return y.cpu() if self.comm_mode == "cpu" else y
+
+    def parameter_rrefs(self):
+        return [rpc.RRef(p) for p in self.parameters()]
+
+
+def measure(comm_mode):
+    # local module on "worker0/cuda:0"
+    lm = MyModule("cuda:0", comm_mode)
+    # remote module on "worker1/cuda:1"
+    rm = rpc.remote("worker1", MyModule, args=("cuda:1", comm_mode))
+    # prepare random inputs
+    x = torch.randn(1000, 1000).cuda(0)
+
+    tik = time.time()
+    for _ in range(10):
+        with autograd.context() as ctx:
+            y = rm.rpc_sync().forward(lm(x))
+            autograd.backward(ctx, [y.sum()])
+    # synchronize on "cuda:0" to make sure that all pending CUDA ops are
+    # included in the measurements
+    torch.cuda.current_stream("cuda:0").synchronize()
+    tok = time.time()
+    print(f"{comm_mode} RPC total execution time: {tok - tik}")
+
+
+class CudaDistributedRPCTest(RpcAgentTestFixture):
+
+    @property
+    def world_size(self):
+        return 2
+
+    @dist_init(setup_rpc=False)
+    def test_cuda_distributed_rpc(self):
+
+        if not torch.cuda.is_available():
+            sys.exit(TEST_SKIPS['no_cuda'].exit_code)
+
+        if torch.cuda.is_available() and torch.cuda.device_count() < 2:
+            sys.exit(TEST_SKIPS['multi-gpu'].exit_code)
```

Comment:
We can use the `skip_if_not_multigpu` and `skip_if_no_gpu` decorators from  `torch.testing._internal.common_distributed` instead

Reply:
Looks like there is something wrong with the `skip_if_not_multigpu` decorator implementation as it is currently skipping all tests. Thanks @gcramer23 for bringing this up!

Reply:
the issue I created https://github.com/pytorch/pytorch/issues/54887

Reply:
As Shen mentioned, adding `@skip_if_lt_x_gpu(2)` and `@skip_if_no_gpu` should work instead

Reply:
added